### PR TITLE
Add nightly-release-binder branch

### PR DESCRIPTION
**Pull request by BetsyMcPhail**
For bionic nightly packaging builds add a step to:
* Edit .binder/Dockerfile to change the "FROM robotlocomotion/drake:latest"
  line to "FROM robotlocomotion/drake:bionic-{date}"
* Commit change
* Push to the refs for `nightly-release-binder`

Relates [#13209](https://github.com/RobotLocomotion/drake/issues/13209)

### CODE REVIEW

**Comment by jamiesnape** [comment]
Once this has merged, need to update any (partial) URLs in drake from `https://mybinder.org/v2/gh/RobotLocomotion/drake/nightly-release` to `https://mybinder.org/v2/gh/RobotLocomotion/drake/nightly-release-binder`. Check for other mentions of `nightly-release` and whether they relate to binder and need changing too.

File: driver/configurations/bazel/step-push-nightly-release-branch.cmake [hunk 1]
```diff
@@ -57,4 +57,45 @@ else()
       append_step_status("PUSHING NIGHTLY RELEASE BRANCH (GIT PUSH)" UNSTABLE)
     endif()
   endif()
+
+  notice("CTest Status: PUSHING NIGHTLY RELEASE BINDER BRANCH")
+
+  # Edit .binder/Dockerfile
+  set(SED_REPLACE_STR
+    "s|^FROM robotlocomotion/drake:latest.*|FROM robotlocomotion/drake:${DISTRIBUTION}-${DATE}|")
+  set(SED_ARGS "-i '${SED_REPLACE_STR}' ${CTEST_SOURCE_DIRECTORY}.binder/Dockerfile")
```
Comment (jamiesnape) [inline driver/configurations/bazel/step-push-nightly-release-branch.cmake:66]: Does `CTEST_SOURCE_DIRECTORY` already have a trailing slash?

File: driver/configurations/bazel/step-push-nightly-release-branch.cmake [hunk 2]
```diff
@@ -57,4 +57,45 @@ else()
...
+        RESULT_VARIABLE DASHBOARD_GIT_PUSH_NIGHTLY_RELEASE_BINDER_BRANCH_RESULT_VARIABLE
+        COMMAND_ECHO STDERR)
+    if(NOT DASHBOARD_GIT_PUSH_NIGHTLY_RELEASE_BINDER_BRANCH_RESULT_VARIABLE EQUAL 0)
+      append_step_status("PUSHING NIGHTLY RELEASE BINDER BRANCH (GIT PUSH)" UNSTABLE)
+    endif()
+  endif()
+
+  # Push to nightly-release-binder branch
+  if(NOT DASHBOARD_UNSTABLE)
+    set(GIT_PUSH_ARGS "push --progress origin HEAD:refs/heads/nightly-release-binder")
```
Comment (jamiesnape) [inline driver/configurations/bazel/step-push-nightly-release-branch.cmake:91]: Do we need `--force`?

**Comment by BetsyMcPhail** [comment]
> Once this has merged, need to update any (partial) URLs in drake from `https://mybinder.org/v2/gh/RobotLocomotion/drake/nightly-release` to `https://mybinder.org/v2/gh/RobotLocomotion/drake/nightly-release-binder`. Check for other mentions of `nightly-release` and whether they relate to binder and need changing too.

Makes sense, I'll get a Drake PR reeady to go

File: driver/configurations/bazel/step-push-nightly-release-branch.cmake [hunk 3]
```diff
@@ -57,4 +57,45 @@ else()
...
+  set(SED_REPLACE_STR
+    "s|^FROM robotlocomotion/drake:latest.*|FROM robotlocomotion/drake:${DISTRIBUTION}-${DATE}|")
+  set(SED_ARGS "-i '${SED_REPLACE_STR}' ${CTEST_SOURCE_DIRECTORY}.binder/Dockerfile")
+  separate_arguments(SED_ARGS_LIST UNIX_COMMAND "${SED_ARGS}")
+  execute_process(COMMAND "sed"  ${SED_ARGS_LIST}
+    WORKING_DIRECTORY "${CTEST_SOURCE_DIRECTORY}"
+    RESULT_VARIABLE DASHBOARD_GIT_PUSH_NIGHTLY_RELEASE_BINDER_BRANCH_RESULT_VARIABLE
+    COMMAND_ECHO STDERR)
+  if(NOT DASHBOARD_GIT_PUSH_NIGHTLY_RELEASE_BINDER_BRANCH_RESULT_VARIABLE EQUAL 0)
+    append_step_status("PUSHING NIGHTLY RELEASE BINDER BRANCH (GIT PUSH)" UNSTABLE)
```
Comment (jamiesnape) [inline driver/configurations/bazel/step-push-nightly-release-branch.cmake:73]: Update the part in parentheses to distinguish between the sed substitution, commit, and push.

File: driver/configurations/bazel/step-push-nightly-release-branch.cmake (hunk 1 above)
Comment (BetsyMcPhail) [reply driver/configurations/bazel/step-push-nightly-release-branch.cmake]: It doesn't appear to.. fixed

File: driver/configurations/bazel/step-push-nightly-release-branch.cmake (hunk 2 above)
Comment (BetsyMcPhail) [reply driver/configurations/bazel/step-push-nightly-release-branch.cmake]: yes

**Comment by BetsyMcPhail** [comment]
Commit on nightly-release-binder looks good the first run (https://drake-jenkins.csail.mit.edu/view/Packaging/job/linux-bionic-unprovisioned-gcc-bazel-nightly-snopt-packaging/667):

```
-FROM robotlocomotion/drake:latest
+FROM robotlocomotion/drake:bionic-20200901
```


But then get error (
https://drake-jenkins.csail.mit.edu/view/Packaging/job/linux-bionic-unprovisioned-gcc-bazel-nightly-snopt-packaging/669)

```
[10:08:26 PM]  remote: error: GH006: Protected branch update failed for refs/heads/nightly-release-binder.        
[10:08:26 PM]  remote: error: Cannot force-push to this protected branch        
[10:08:26 PM]  To github.com:RobotLocomotion/drake.git
[10:08:26 PM]   ! [remote rejected]     HEAD -> nightly-release-binder (protected branch hook declined)
[10:08:26 PM]  error: failed to push some refs to 'git@github.com:RobotLocomotion/drake.git'
```

**Comment by jamiesnape** [comment]
Try again now. That branch name was caught by a glob in the list of protected branches. I have (hopefully) overridden it to allow force pushes.

**Comment by BetsyMcPhail** [comment]
Thanks @jamiesnape - looks good now --> https://drake-jenkins.csail.mit.edu/view/Packaging/job/linux-bionic-unprovisioned-gcc-bazel-nightly-snopt-packaging/671/console

**Pull request merged**